src/arch/x86_64/CodeGen.zig

```diff
@@ -3125,7 +3137,6 @@ fn setRegOrMem(self: *Self, ty: Type, loc: MCValue, val: MCValue) !void {
 fn genSetStack(self: *Self, ty: Type, stack_offset: u32, mcv: MCValue) InnerError!void {
     switch (mcv) {
         .dead => unreachable,
-        .ptr_stack_offset => unreachable,
```

Comment:
Hmm, as far as I know, `genSetStack`in a `ptr_stack_offset` shouldn't be possible, right? Just like `genSetStack`in an `immediate`. `ptr_stack_offset` represents not the value at that stack offset, but the stack offset itself.

What exactly triggers a `genSetStack` for `ptr_stack_offset`?

Reply:
> Just like `genSetStack`ing an `immediate`.

Oh, nevermind. That's perfectly possible. But shouldn't the code paths for `stack_offset` and `ptr_stack_offset` be different?

Reply:
So the codepaths share the bit until abi_size is within 8 bytes (for ptr to stack offset, that's obv always 8 bytes), but I take your point that having this more explicit will not be as confusing to newcomers. I will change it so that it's easier to reason about, and when we have enough implemented, we can start trimming it down and refactoring.